<!-- 
###########################################################################################
###########################################################################################
## Projeto: Nanook UFC
## Autor: Kaio Douglas Teófilo Rocha
## Email: kaiodtr@gmail.com
###########################################################################################
## Arquivo: Launcher Básico do Robô 
## Revisão: 1 [02/03/2017]
###########################################################################################
###########################################################################################
 -->

<launch>

	<!-- Nó Controlador de Base -->

	<node pkg="nanook_ufc" type="base_controller.py" name="base_controller" output="screen">

		<!-- Frequência principal -->
		<param name="base_frequency" type="double" value="50" />
		
		<!-- Dimensões do robô -->
		<param name="wheel_diameter" type="double" value="0.17" />
		<param name="wheel_separation" type="double" value="0.405" />

		<!-- Driver do controlador da base (FRDM K64F) -->
		<param name="base_serial_port" type="string" value="/dev/ttyACM0" />
		<param name="base_baud_rate" type="int" value="115200" />	

		<!-- Pose inicial -->
		<param name="x_0" type="double" value="0.0" />
		<param name="y_0" type="double" value="0.0" />
		<param name="theta_0" type="double" value="1.570796" />

		<!-- Nomenclatura dos sistemas de coordenadas -->
		<param name="base_frame_id" type="string" value="base_link" />
		<param name="odom_frame_id" type="string" value="odom" />

	</node>

	<!-- Nó Publicador de Transformações Fixas -->

	<node pkg="nanook_ufc" type="fixed_tf_broadcaster.py" name="fixed_tf_broadcaster" >

		<!-- Frequência de publicaçao -->
		<param name="publishing_frequency" type="double" value="25" />

		<!-- Posição do laser (em relação ao centro da base) -->
		<param name="laser_x" type="double" value="0.2" />	
		<param name="laser_y" type="double" value="0.0" />	
		<param name="laser_z" type="double" value="0.4" />

		<!-- Nomenclatura dos sistemas de coordenadas -->
		<param name="base_frame_id" type="string" value="base_link" />
		<param name="laser_frame_id" type="string" value="laser_scanner" />

	</node>

</launch>